web/src/components/planner/products/BuildingGroup.vue

```diff
@@ -0,0 +1,330 @@
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { defineProps, inject } from 'vue'
+  import {
+    BuildingGroup,
+    Factory,
+    FactoryItem,
+    FactoryPowerProducer,
+    GroupType,
+  } from '@/interfaces/planner/FactoryInterface'
+  import eventBus from '@/utils/eventBus'
+  import { getPartDisplayName } from '@/utils/helpers'
+  import { useGameDataStore } from '@/stores/game-data-store'
+  import { increaseProductQtyViaBuilding } from '@/utils/factory-management/products'
+  import { useDisplay } from 'vuetify'
+  // import { updateGroupParts } from '@/utils/factory-management/productBuildingGroups'
+  import { formatNumberFully, formatPower } from '@/utils/numberFormatter'
+  import { PowerItem } from '@/interfaces/Recipes'
+
+  const updateFactory = inject('updateFactory') as (factory: Factory) => void
+  const gameData = useGameDataStore().getGameData()
+
+  // const timeout: NodeJS.Timeout | null = null
+  let updatingPart: string
+
+  const { lgAndDown, lgAndUp } = useDisplay()
+
+  const props = defineProps<{
+    factory: Factory
+    group: BuildingGroup
+    item: FactoryItem | FactoryPowerProducer
+    building: string // Building name
+  }>()
+
+  const updateGroup = (group: BuildingGroup) => {
+    if (group.buildingCount === 0 || isNaN(group.buildingCount) || group.buildingCount === null) {
+      eventBus.emit('toast', {
+        message: 'Building count must be a positive number.',
+        type: 'warning',
+      })
+      group.buildingCount = 1
+      return
+    }
+
+    // Ensure the building count is a whole number
+    if (group.buildingCount % 1 !== 0) {
+      eventBus.emit('toast', {
+        message: 'Building count must equal to a whole number. If you need a single building clocked, create a new building group and adjust it\'s clock.',
+        type: 'error',
+        timeout: 5000,
+      })
+      group.buildingCount = Math.floor(group.buildingCount)
+    }
+
+    if (props.item.buildingGroups.length === 1) {
+      // Since we have edited the buildings in the group, we now need to edit the product's building requirements.
+      if (props.group.type === GroupType.Product) {
+        const subject = props.item as FactoryItem
+        subject.buildingRequirements.amount = group.buildingCount
+        increaseProductQtyViaBuilding(subject, gameData)
+      } else if (props.group.type === GroupType.Power) {
+        const subject = props.item as FactoryPowerProducer
+        subject.buildingCount = 1
+      } else {
+        throw new Error('Invalid type')
+      }
+    }
+
+    // If the user is trying to use more than .0001 precision for overclock, truncate it and alert them.
+    const clock = group.overclockPercent.toString().split('.')[0]
+    const precision = group.overclockPercent.toString().split('.')[1]
+    if (precision?.length > 4) {
+      // Truncate the overclock to 4 decimal places.2255255255525555
```

Comment:
What are the numbers here in this comment?

Reply:
Not quite sure... thanks for pointing it out 
```suggestion
      // Truncate the overclock to 4 decimal places
```